torchdynamo/variables/tensor.py

```diff
@@ -145,32 +148,26 @@ def context():
                             lambda: cls.run_proxy(proxy, args, kwargs, nnmodule)
                         )
                 except RuntimeError as e:
-                    raise TorchRuntimeError() from e
+                    if use_fake_tensors and isinstance(e, DataDependentOutputException):
+                        if (
+                            config.capture_scalar_outputs
+                            and proxy.node.target == "item"
+                        ):
+                            example_value = torch.zeros(
+                                size=(), dtype=args[0].dtype
+                            ).item()
+                        else:
+                            unimplemented("data dependent operator")
+                    elif use_fake_tensors and isinstance(
+                        e, DynamicOutputShapeException
+                    ):
+                        unimplemented("dynamic shape operator")
```

Comment:
Is there any data from the exception we can plumb here for a nicer message?

Reply:
we can print the dynamic shape operator, yea